lib/semian/net_http.rb

```diff
@@ -87,14 +88,24 @@ def disabled?
       raw_semian_options.nil?
     end
 
+    def fatal_server_errors?
+      raw_semian_options.fetch(:fatal_server_errors)
+    end
+
     def connect
       return raw_connect if disabled?
       acquire_semian_resource(adapter: :http, scope: :connection) { raw_connect }
     end
 
     def request(req, body = nil, &block)
       return raw_request(req, body, &block) if disabled?
-      acquire_semian_resource(adapter: :http, scope: :query) { raw_request(req, body, &block) }
+      acquire_semian_resource(adapter: :http, scope: :query) do
+        result = raw_request(req, body, &block)
+
+        result.error! if result.is_a?(::Net::HTTPServerError) && fatal_server_errors?
```

Comment:
@sirupsen are you suggesting that we raise the exception somewhere else? Maybe here: https://github.com/Shopify/semian/blob/4e61453710bf0dc8dc5de35f83bfee31f3d92742/lib/semian/adapter.rb#L32? 

The rationale of putting it there is that we wanted to limit this behaviour to Net::HTTP requests. 

Are there any other alternatives I'm missing?